### PR TITLE
feat: trim country from VAT number if country matches the selected one (issue #56032)

### PR DESCRIPTION
**Pull request by fabz1001**
#### Changes proposed in this Pull Request

Leading country to be trimmed from VAT number if it matches the selected one.

#### Testing instructions

1. Visit `/me/purchases/vat-details`
1. Complete the form and prepend the country code to the VAT number
2. Submit the form
3. Check the request payload

vat_details

Related to #56032

### CODE REVIEW

**Review by sirbrillig — COMMENTED** [review]
This works great! Thank you! The only thing that worries me is that the trimming happens live, while typing. This is actually really efficient but it does make me wonder if it might be confusing UX? Here's what I mean:

1. User selects "UK"
2. User begins typing UK1234: "U... K..." (field contents vanishes) "oh! something went wrong I guess. Let me try again."

Maybe it would be better to do the trimming just after the submit button is pressed. What do you think?

Alternatively we could just defer the trimming to wait until the user stops typing (and maybe also only do it if there's more than three characters in the field).

Yet another alternative would be to show some affordance when the trimming happens, but that sounds like more work than just changing the timing.

**Comment by toncijajic** [comment]
If I can recommend the best-case UX behavior here, it would be to trim the leading code in the call made to VIES API (I'm guessing that's what's used) while keeping the user's input intact. That's the least confusing scenario.

**Comment by fabz1001** [comment]
I believe you both are right. As much as I like some movement in the UI, the best thing to do here would probably be doing the trimming at form submission. 
I'm going to update the PR accordingly tomorrow. Thanks! 🙂

**Comment by fabz1001** [comment]
@sirbrillig I have updated the PR. Thanks 🙂

**Review by sirbrillig — APPROVED** [review]
Works as described, and the code looks good to me.

I tried submitting an ID prefixed with the country code and while it appeared to be unchanged, the network request was made with just the ID. Submitting just the ID is unaffected.

**Comment by sirbrillig** [comment]
Rebasing to try to fix weird e2e test failures (`Error response from daemon: manifest for registry.a8c.com/calypso/ci-desktop:latest not found: manifest unknown: manifest unknown`)
